docs/pages/access-controls/sso/google-workspace.mdx

```diff
@@ -170,87 +149,225 @@ correct Google Workspace plan.
+  - claim: groups
+    roles:
+    - auditor
+    value: auditor@example.com
+  - claim: groups
+    roles:
+    - access
+    value: teleport-developers@example.com
+  client_id: <CLIENT-ID>
+  client_secret: <CLIENT-SECRET>
+  display: Google
+  google_admin_email: <GOOGLE-WORKSPACE-ADMIN-EMAIL>
+  google_service_account: |2-
+
+    {
+      "type": "service_account",
+      ...
+      "universe_domain": "googleapis.com"
+    }
+  issuer_url: https://accounts.google.com
+  redirect_url: https://example.teleport.sh:443/v1/webapi/oidc/callback
+version: v3
+```
+
+</TabItem>
+<TabItem scope={["oss", "enterprise"]} label="Uploaded JSON file">
+
+Use this method for single self-hosted Teleport Auth instances, or when you can
+easily and reliably make the JSON file available to all hosts running the Auth
+service.
+
+```code
+$ tctl sso configure oidc --preset google --id <CLIENT-ID> \
+--secret <CLIENT-SECRET> \
+--google-acc-uri <PATH/TO/SERVICE-ACCOUNT-KEY>.json \
+--claims-to-roles groups,auditor@example.com,auditor \
+--claims-to-roles groups,teleport-developers@example.com,access \
+--google-admin=<GOOGLE-WORKSPACE-ADMIN-EMAIL> > gworkspace-connector.yaml
+```
+
+The file created will resemble:
+
+```yaml
+kind: oidc
+metadata:
+  name: google
+spec:
+  claims_to_roles:
+  - claim: groups
+    roles:
+    - auditor
+    value: auditor@example.com
+  - claim: groups
+    roles:
+    - access
+    value: teleport-developers@example.com
+  client_id: <CLIENT-ID>
+  client_secret: <CLIENT-SECRET>
+  display: Google
+  google_admin_email: <GOOGLE-WORKSPACE-ADMIN-EMAIL>
+  google_service_account_uri: /PATH/TO/SERVICE-ACCOUNT-KEY.json
+  issuer_url: https://accounts.google.com
+  redirect_url: https://teleport.example.com/v1/webapi/oidc/callback
+version: v3
+```
+
+</TabItem>
 </Tabs>
 
-The email that you set for `google_admin_email` **must** be the email address of a user that has permission to list all groups, users, and group membership in your Google Workspace account. This user will generally need super admin or group admin privileges.
+<Admonition
+  type="tip"
+  title="Copy/Paste Tip"
+>
+  Be sure to remove < > brackets around tokens in the sample configuration above.
+  For example, replace `<GOOGLE-WORKSPACE-ADMIN-EMAIL>` with `admin@yourdomain.com`.
+</Admonition>
+
+The email that you set for `google_admin_email` **must** be the email address of
+a user that has permission to list all groups, users, and group membership in
+your Google Workspace account. This user will generally need super admin or group
+admin privileges.
+
+Do not use the email of the service account for `google_admin_email`. The
+configuration display will look the same, but the service account will not have
+the required domain-wide delegation.
+
+The `client_id` field must be the unique ID number captured from the Google Cloud
+Platform UI. An indicator that this is misconfigured is if you see "invalid Google
+Workspace credentials for scopes [...]" in your log.
```

Comment:
Which log?

Reply:
I don't know. While this is green because of the formatting changes, this content predates my updates. Having done it right :puts-shades-on: I didn't encounter this error in any log.

Reply:
It's probably:

```suggestion
Workspace credentials for scopes [...]" in your Teleport logs or the `tctl` output.
```

But ping @Tener to confirm.

Reply:
Double-checking with @Tener. If it's the Auth Service log, this won't be available to cloud-hosted Teleport admins

Reply:
You reminded me that audit logs are accessible in the Web UI, where I found an example of this error. So to solve:

```suggestion
Workspace credentials for scopes [...]" in your audit log, accessible from the
Auth service host or the **Management** section of the web UI.
```